src/Employer.Web/Configuration/ConfigurationExtensions.cs

```diff
@@ -98,7 +100,7 @@ public static void AddAuthenticationService(this IServiceCollection services, Au
         private static Task PopulateAccountsClaim(Microsoft.AspNetCore.Authentication.OpenIdConnect.TokenValidatedContext ctx, IGetAssociatedEmployerAccountsService accountsSvc)
         {
             var userId = ctx.Principal.Claims.First(c => c.Type.Equals(EmployerRecruitClaims.IdamsUserIdClaimTypeIdentifier)).Value;
-            var accounts = accountsSvc.GetAssociatedAccounts(userId);
+            var accounts = accountsSvc.GetAssociatedAccounts(userId).Result;
```

Comment:
I think we may be able to async the calling function so we don't have to wait for the result. options.Events.OnTokenValidated = async (ctx) => PopulateAccountsClaim(ctx, accountsSvc);